server/src/test/java/org/elasticsearch/action/admin/cluster/snapshots/create/CreateSnapshotResponseTests.java

```diff
@@ -64,6 +66,13 @@ protected CreateSnapshotResponse createTestInstance() {
         boolean globalState = randomBoolean();
 
         return new CreateSnapshotResponse(
-            new SnapshotInfo(snapshotId, indices, startTime, reason, endTime, totalShards, shardFailures, globalState));
+            new SnapshotInfo(snapshotId, indices, startTime, reason, endTime, totalShards, shardFailures,
+                globalState, SnapshotInfoTests.randomUserMetadata()));
+    }
+
+    @Override
+    protected Predicate<String> getRandomFieldsExcludeFilter() {
```

Comment:
Can you elaborate on this? Can you give an example of the CreateSnapshotResponse and test that will fail.

Reply:
I'll add some explanation to the comment here as well, but this is due to the behavior of `AbstractXContentTestCase` when `supportsUnknownFields` is set to `true` as it is in this case.  One of the tests it runs creates a random instance, serializes it, adds random fields to the xcontent, then deserializes it and asserts that the deserialized instance is equal to the original instance.  It adds the new, random fields at not just the base level, but also inside any object fields, unless you tell it not to via this method.

Let's say the generated CreateSnapshotResponse is serialized as something like this:
```json
{
  "snapshot": {
    "snapshot": "my-snapshot",
    "indices": ["test1", "test2"],
    "_meta": {
      "my-field": "my-value"
    },
    // [...other fields elided for brevity...]
  }
}
```

This is then modified by `AbstractXContentTestCase` to add some random fields (because we told it this object supports unknown fields), so it looks like then when we go to parse it:

```json
{
  "snapshot": {
    "nrkw": "asdfhqriufn",
    "snapshot": "my-snapshot",
    "p3mt": "pqui3ghp39jf",
    "indices": ["test1", "test2"],
    "_meta": {
      "pwmx": "qwgre",
      "my-field": "my-value"
    },
    // [...other fields elided for brevity...]
  }
}
```

Now the fields that were added in the top level (`nrkw` and `p3mt` in this example) are ignored, because the parser ignores them. But since `_meta` is a `Map<String, Object>`, it just adds those keys to the map.

Now, when the deserialized version is checked for equality against the original, it's no longer equal, because the `Map` object has a new entry in it that wasn't there before.

Overriding this method to prevent `AbstractXContentTestCase` from adding randomly generated fields to `_meta` avoids this problem.

Reply:
Can you extend the comment please?

Reply:
Thanks, I forgot about this one apparently.  How does the comment look now? I tried to trim down the explanation from the one above while still making it clear what's going on.

I also noticed that the predicate wasn't actually correct due to a misunderstanding on my part, as written it was effectively just disabling unknown fields.  I've changed them to work as intended.